### PR TITLE
Add CUDA and ROCM docs on website

### PR DESCRIPTION
**Pull request by raikonenfnu**

### CODE REVIEW

File: docs/website/docs/deployment-configurations/gpu-cuda-rocm.md [hunk 1]
```diff
@@ -0,0 +1,203 @@
+# CUDA and ROCM GPU HAL Driver
+
+A lot of AI/ML workload uses GPU to accelerate execution. IREE can accelerate model execution on GPUs through CUDA for NVIDIA hardwares, and ROCm for AMD hardwares. Due to the similarity of CUDA and ROCM APIs and infrastructure, CUDA and ROCM backend shares many features and implementation in IREE. In IREE compiler level, CUDA and ROCM bothg go through LLVMGPU pipeline, but CUDA generate ptx, while ROCM generate hsaco. In the IREE runtime level, the HAL layer for ROCM mirrors the one on CUDA, except for the command graph. Where CUDA has direct, stream, and graph command buffer, and ROCM has only direct command buffer.
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/gpu-cuda-rocm.md:3]: a few grammar and capitalization nits

```suggestion
IREE can accelerate model execution on NVIDIA GPUs using CUDA and on AMD GPUs using ROCm. Due to the similarity of CUDA and ROCm APIs and infrastructure, the CUDA and ROCm backends share much of their implementation in IREE. The IREE compiler uses a similar GPU code generation pipeline for each, but generates PTX for CUDA and hsaco for ROCm. The IREE runtime HAL driver for ROCm mirrors the one for CUDA, except for the command graph - where CUDA has "direct", "stream", and "graph" command buffers, and ROCM has only "direct" command buffers.
```

File: docs/website/docs/deployment-configurations/gpu-cuda-rocm.md [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+
+## Prerequisites
+
+In order to use CUDA or ROCM to drive the GPU, you need to have a functional CUDA or ROCM
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/gpu-cuda-rocm.md:7]: nit: ROCm

```suggestion
In order to use CUDA or ROCm to drive the GPU, you need to have a functional CUDA or ROCm
```

File: docs/website/docs/deployment-configurations/gpu-cuda-rocm.md [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+[TensorFlow importer][tf-import]. Then,
+
+#### Compile using the command-line
+
+Let `iree_input.mlir` be the model's initial MLIR representation generated by
+IREE's TensorFlow importer. We can now compile them for each GPU by running the following command:
+
+=== "Nvidia/CUDA"
+
+``` shell hl_lines="3 4"
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/gpu-cuda-rocm.md:118]:
```suggestion
``` shell hl_lines="3 5"
```

highlight line 5 also?

this is how the highlighting appears on the site:
image

docs: https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#highlighting-specific-lines

File: docs/website/docs/deployment-configurations/gpu-cuda-rocm.md [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+
+=== "AMD/ROCM"
+
+``` shell hl_lines="3 4"
+iree/tools/iree-translate \
+    -iree-mlir-to-vm-bytecode-module \
+    -iree-hal-target-backends=rocm \
+    -iree-rocm-target-chip=<...> \
+    -iree-rocm-link-bc=true \
+    -iree-rocm-bc-dir=<...> \
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/gpu-cuda-rocm.md:147]: highlight all rocm lines?

```suggestion
``` shell hl_lines="3 6"
iree/tools/iree-translate \
    -iree-mlir-to-vm-bytecode-module \
    -iree-hal-target-backends=rocm \
    -iree-rocm-target-chip=<...> \
    -iree-rocm-link-bc=true \
    -iree-rocm-bc-dir=<...> \
```

File: docs/website/docs/deployment-configurations/gpu-cuda-rocm.md [hunk 5]
```diff
@@ -0,0 +1,203 @@
...
+    iree_input.mlir -o mobilenet-rocm.vmfb
+```
+
+Note ROCM Bitcode Dir(`iree-rocm-bc-dir`) path is required. If the system you are compiling IREE in has ROCM installed, then the default value of `/opt/rocm/amdgcn/bitcode` will usually suffice. If you intend on building ROCM compiler in a non-ROCM capable system, please set `iree-rocm-bc-dir` to the absolute path where you might have saved the amdgcn bitcode.
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/gpu-cuda-rocm.md:151]: nit: ROCm
```suggestion
Note ROCm Bitcode Dir(`iree-rocm-bc-dir`) path is required. If the system you are compiling IREE in has ROCm installed, then the default value of `/opt/rocm/amdgcn/bitcode` will usually suffice. If you intend on building ROCm compiler in a non-ROCm capable system, please set `iree-rocm-bc-dir` to the absolute path where you might have saved the amdgcn bitcode.
```

File: docs/website/docs/deployment-configurations/gpu-cuda-rocm.md [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+    --module_file=mobilenet-cuda.vmfb \
+    --entry_function=predict \
+    --function_input="1x224x224x3xf32=0"
+```
+
+=== "AMD/ROCM"
+
+``` shell hl_lines="2"
+iree/tools/iree-run-module \
+    --driver=cuda \
```
Comment (ScottTodd) [inline docs/website/docs/deployment-configurations/gpu-cuda-rocm.md:184]: should this use `--driver=rocm`?
```suggestion
    --driver=rocm \
```

**Review by ScottTodd — COMMENTED** [review]
Generally LGTM 👍 

a few grammar nits and other small suggestions

**Comment by raikonenfnu** [comment]
Awesome! Thanks for the corrections Scott! :)

**Review by ScottTodd — APPROVED** [review]

**Pull request merged**